server/index.js

```diff
@@ -0,0 +1,149 @@
+const express = require('express');
+const db = require('../database/index');
+const bodyParser = require('body-parser');
+const morgan = require('morgan');
+
+const app = express();
+const PORT = 3002;
+
+app.use(bodyParser.json());
+app.use(morgan('dev'));
+
+app.get('/books/:id', async (req, res) => {
+  const id = req.params.id;
+  if(!/^\d+$/.test(id))
```

Comment:
Nice idea to test the id. I understand this is a regular expression test function but could you explain what exactly the regular expression is testing?

Reply:
^ means the start of a string. \d means 0-9. + means repeat that 1 or more times, so 1 or 232 or 923923, and $ means the end of the string. So basically, I'm testing the whole string is build with numbers